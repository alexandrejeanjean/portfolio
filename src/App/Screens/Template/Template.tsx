import React, { useEffect, useState } from 'react'
import { useLocation } from 'react-router-dom'
import Header from '../../Shared/Header/Header'
import Navigation from '../../Core/Navigation/Navigation'
import MobileNavigation from '../../Core/MobileNavigation/MobileNavigation'
import BrainHead from '../../Assets/BrainHead'

type Props = { children: React.ReactNode }

const Template = ({ children }: Props) => {
  let [path, setNewPath] = useState('')
  let location = useLocation()

  useEffect(() => {
    if (location.pathname !== path) {
      let newPath = location.pathname
      setNewPath(newPath)
    }
  }, [location, path])

  const setBackground = () => {
    if (['/services', '/a-propos', '/projets'].indexOf(path) > -1) {
      return <BrainHead className='about-bubble' />
    }
    return null
  }

  return (
    <>
      <Header />
      <Navigation path={path} />
      <MobileNavigation path={path} />
      {children}
      {setBackground()}
    </>
  )
}

export default Template
