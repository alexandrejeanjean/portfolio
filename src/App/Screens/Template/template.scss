.bg-bubble {
  position: fixed;
  top: 10%;
  right: -30%;
  margin: auto;
  width: 100%;
  z-index: -1;
}

@media screen and (max-width: 768px) {
  .bg-bubble {
    display: none;
  }
}
