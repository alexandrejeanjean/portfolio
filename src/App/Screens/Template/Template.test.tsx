import React from 'react'
import { shallow } from 'enzyme'
import Template from './Template'

let wrapper: any

beforeEach(() => {
  wrapper = shallow(<Template children />)
})

describe('Template view', () => {
  it('should render Template component correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
