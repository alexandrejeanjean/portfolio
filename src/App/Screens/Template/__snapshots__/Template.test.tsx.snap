// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template view should render Template component correctly 1`] = `
<Fragment>
  <Header />
  <Navigation />
  <Component />
</Fragment>
`;
